UserService/UserService.Handlers/PutModifyRegistrationPageTwoHandler.cs

```diff
@@ -6,23 +6,44 @@
 using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
+using UserService.Core.Interfaces.Services;
+using HelpMyStreet.Contracts.CommunicationService.Request;
 
 namespace UserService.Handlers
 {
     public class PutModifyRegistrationPageTwoHandler : IRequestHandler<PutModifyRegistrationPageTwoRequest, PutModifyRegistrationPageTwoResponse>
     {
         private readonly IRepository _repository;
+        private readonly ICommunicationService _communicationService;
 
-        public PutModifyRegistrationPageTwoHandler(IRepository repository)
+        public PutModifyRegistrationPageTwoHandler(IRepository repository, ICommunicationService communicationService)
         {
             _repository = repository;
+            _communicationService = communicationService;
         }
 
-        public Task<PutModifyRegistrationPageTwoResponse> Handle(PutModifyRegistrationPageTwoRequest request, CancellationToken cancellationToken)
+        public async Task<PutModifyRegistrationPageTwoResponse> Handle(PutModifyRegistrationPageTwoRequest request, CancellationToken cancellationToken)
         {
             int response = _repository.ModifyUserRegistrationPageTwo(request.RegistrationStepTwo);
 
-            return Task.FromResult(new PutModifyRegistrationPageTwoResponse() { ID = response});
+            var user =_repository.GetUserByID(response);
+
+            if (user != null)
+            {
+                bool addedContact = await _communicationService.PutNewMarketingContactAsync(new PutNewMarketingContactRequest()
```

Comment:
Again, should probably do something with the response...  Ideally we'd log the error but not interrupt the flow in this scenario, but I'm not sure we have somewhere to log it currently, where we'd notice that it had happened.